documentation/API.md

```diff
@@ -352,6 +383,88 @@ For displaying custom data (e.g. Users, Products, Orgs). User's JWT determines w
+
+See examples in the core/demo templates for how to configure action previews.
+
+##### REQUEST parameters:
+
+- `applicationId`
+- `reviewId` (one of either `applicationId` or `reviewId` must be provided)
+- `applicationDataOverride` an object containing data to override the generated applicationData. For example, when simulating a decision, we would override the `reviewData.latestDecision.decision` field with the hypothetical decision value and then the action would be "previewed" as though that were the actual applicationData it uses.
+
+
+##### RESPONSE Body (example):
+
+```JSON
+{
+    "displayData": [
+        {
+            "type": "NOTIFICATION",
+            "status": "SUCCESS",
+            "displayString": "Congratulations, application S-UUR-0001 has been approved",
+            "text": "## Your product registration license ...",
+            "errorLog": null
+        }
+    ],
+    "actionsOutput": [
+        {
+            "action": "sendNotification",
+            "status": "SUCCESS",
+            "output": {
+                "notification": {
+                    "id": 2,
+                    "user_id": 9,
+                    "application_id": 239,
+                    "review_id": 3,
+                    "email_recipients": "carl@msupply.foundation",
+                    "subject": "Congratulations, application S-UUR-0001 has been approved",
+                    "message": "## Your product registration license ...",
+                    "attachments": [],
+                    "email_sent": false,
+                    "is_read": false
+                }
+            },
+            "errorLog": null
+        }
+    ]
+}
+```
+
+`displayData` is used by the front-end to present the Preview results in the UI. `actionsOutput` contains the raw output of each action that ran.
+
+
+
+### Extend application deadline
+
+POST: `/extend-application`
+
+Endpoint for extending a deadline associated with an application. Currently, the main use case is for extending an applicant deadline for responding to a request for changes (event code: `applicantDeadline`), but in theory could be used for extending other types of deadlines as well.
+
+It works by finding an event in the `trigger_schedule` table with matching `applicationId` and `eventCode` and then extending the `time_scheduled` by the specified amount. It also simulates a trigger `ON_EXTEND` which can be used to trigger other events in the application (such as resetting the outcome from "EXPIRED" back to "PENING" for example).
```

Comment:
I'm not entirely sure on this sentence. Maybe could use some update to be clearer.

Particularly this sentence: 
> It also simulates a trigger `ON_EXTEND` which can be used to trigger other events in the application (such as resetting the outcome from "EXPIRED" back to "PENING" for example).

Reply:
Cool -- how's this? 9a6ad9dd

Reply:
I think it's easier to understand. But we should probably go through all the documentation at some point and refresh, specially when translating it to Portuguese. I think it's hard to understand a few concepts when it's very particular to one action.

Reply:
> specially when translating it to Portuguese

This is developer reference docs -- would we ever translate this? (If I were writing it for general access, I'd write it completely differently, I'm mainly writing this stuff down so _I_ can remember it heh)